### PR TITLE
doc: release notes: ARC

### PR DESCRIPTION
**Pull request by evgeniy-paltsev**
2.5 release notes for ARC related changes

### CODE REVIEW

**Review by nashif — APPROVED** [review]

**Review by ruuddw — APPROVED** [review]

File: doc/releases/release-notes-2.5.rst [hunk 1]
```diff
@@ -147,6 +147,16 @@ Architectures
 
 * ARC
 
+  * Fixed execution on HS with one interrupt bank and fast interrupts enabled
```
Comment (abrodkin) [inline doc/releases/release-notes-2.5.rst:150]: I'd use `ARC HS` as a more typical use of the ARC core name. The same goes to "fast interrupts" which might be useful to also mentioned like `FIRQ`. I.e. something like this:

> Fixed execution on ARC HS with one interrupt bank and fast interrupts (FIRQ) enabled

File: doc/releases/release-notes-2.5.rst [hunk 2]
```diff
@@ -182,6 +192,11 @@ Boards & SoC Support
 
 * Changes for ARC boards:
 
+  * Added icount support for ARC QEMU boards
+  * Added MWDT compiler options for HSDK board
+  * Accommodated upstream OpenOCD for ARC
+  * Added missing taps into JTAG chain for HSDK board
```
Comment (abrodkin) [inline doc/releases/release-notes-2.5.rst:198]: for the dual-core configuration of the HSDK

File: doc/releases/release-notes-2.5.rst [hunk 3]
```diff
@@ -182,6 +192,11 @@ Boards & SoC Support
 
 * Changes for ARC boards:
 
+  * Added icount support for ARC QEMU boards
+  * Added MWDT compiler options for HSDK board
+  * Accommodated upstream OpenOCD for ARC
```
Comment (abrodkin) [inline doc/releases/release-notes-2.5.rst:197]: "Accommodated" -> "Switched to"?

**Review by abrodkin — CHANGES_REQUESTED** [review]
Just a very minor wording complaints.

**Review by abrodkin — APPROVED** [review]

**Pull request merged**